src/rpc/misc.cpp

```diff
@@ -383,22 +383,22 @@ UniValue validateaddress(const UniValue& params, bool fHelp)
     LOCK(cs_main);
 #endif
 
-    CBitcoinAddress address(params[0].get_str());
-    bool isValid = address.IsValid();
+    bool isValid = IsValidDestinationString(params[0].get_str());
 
     UniValue ret(UniValue::VOBJ);
     ret.push_back(Pair("isvalid", isValid));
     if (isValid) {
-        CTxDestination dest = address.Get();
-        std::string currentAddress = address.ToString();
+        std::string currentAddress = params[0].get_str();
+        bool isStakingAddress = false;
+        CTxDestination dest = DecodeDestination(currentAddress, isStakingAddress);
```

Comment:
to avoid the double decoding here, might make sense to do `CTxDestination dest = DecodeDestination(params[0].get_str(), isStakingAddress)` at the beginning here, then define `bool isValid = IsValidDestination(dest)`, and exit early if `isValid` is false.